server/src/main/java/org/apache/cassandra/sidecar/CassandraSidecarDaemon.java

```diff
@@ -88,5 +78,55 @@ private static boolean close(Server app)
         }
         return false;
     }
+
+    private static Path determineConfigPath()
+    {
+        String yamlConfigurationPath = System.getProperty("sidecar.config");
+        Path confPath;
+        if (yamlConfigurationPath != null)
+        {
+            try
+            {
+                confPath = Paths.get(new URI(yamlConfigurationPath));
+            }
+            catch (Throwable e)
+            {
+                throw new IllegalArgumentException("Invalid URI: " + yamlConfigurationPath, e);
+            }
+            return ensurePathExists(confPath);
+        }
+
+        confPath = Paths.get("conf/sidecar.yaml");
+        if (Files.exists(confPath))
+        {
+            return confPath;
+        }
+
+        // Try to get the file from the resources directory as fallback
+        try
+        {
+            confPath = Paths.get(Thread.currentThread()
+                                       .getContextClassLoader()
+                                       .getResource("dist/conf/sidecar.yaml")
+                                       .toURI());
+        }
+        catch (Throwable e)
+        {
+            throw new RuntimeException("Unable to obtain resource: dist/conf/sidecar.yaml", e);
+        }
+
+        ensurePathExists(confPath);
```

Comment:
Does it work if the yaml file is in the jar file? Maybe add a test for that.

Reply:
We shouldn't be putting yaml configs into the JAR.

Reply:
Are you suggesting excluding the `sidecar.yaml` from the jar? It is included, as of now.

Reply:
Yes.  Never put config files in the JAR.

A top level `conf` dir would be ideal, because then we can do this in the Gradle config:

```
distributions {
    main {
        // Include the "packer" directory in the distribution
        contents {
            from("conf") {
                into("conf")
            }
        }
    }
}
```

And we have consistent directory structure between this and Cassandra itself, the jar files are excluded, and is set up for the distZip etc to all work the same way as the dev repo.